### PR TITLE
Improve formatting of docstrings in the C API

### PR DESCRIPTION
**Pull request by odow**
This is a very minor thing and I haven't finished yet, but I though I'd open this to start with.

I was working on https://github.com/jump-dev/HiGHS.jl/pull/152 and got annoyed at the inconsistency in capitalization and punctuation in the docstrings that were coming through.

Ideally, I think we should aim for the Gurobi API documentation: https://www.gurobi.com/documentation/10.0/refman/c_setdblattr.html.

### CODE REVIEW

File: src/interfaces/highs_c_api.h [hunk 1]
```diff
@@ -211,80 +213,80 @@ void* Highs_create(void);
...
 
 /**
- * Return the HiGHS major version number
+ * Return the HiGHS major version number.
  *
- * @returns the HiGHS major version number
+ * @returns The HiGHS major version number.
  */
-HighsInt Highs_versionMajor();
+HighsInt Highs_versionMajor(void);
```
Comment (odow) [inline src/interfaces/highs_c_api.h]: Should I make this a separate PR? I got a warning parsing this in Julia.

Comment (jajhall) [reply src/interfaces/highs_c_api.h]: No need for a separate PR. I've no idea what this `void` does, anyway

**Comment by jajhall** [comment]
Thanks for all this @odow . I guess that you need proper sentences in the docstrings for subsequent appearance in documentation.

Are the non-comment changes limited to inserting `void` into some method parameters? They seem to be, as I skimmed through everything. If so - and the CI tests pass anyway - I'm happy to approve the PR for whenever you's happy to merge.

**Comment by jajhall** [comment]
As for aiming for the Gurobi API documentation: https://www.gurobi.com/documentation/10.0/refman/c_setdblattr.html, there's no chance of us giving this level of web-based documentation in the short or medium term. Or do you just mean that the C API docstrings should yield this if processed appropriately?

We've made great strides on [documentation](https://ergo-code.github.io/HiGHS/) in the past few weeks, but C and C++ users will have to read header files to get all the details for the foreseeable future

**Comment by odow** [comment]
> Are the non-comment changes limited to inserting void into some method parameters? 

Yes. I'll move to a separate PR so it's easier to track in future if anyone goes looking.

> Or do you just mean that the C API docstrings should yield this if processed appropriately?

Yeah, I meant the format of the docstrings, not that we need a website with documentation.

The docstrings in HiGHS.jl show up as:

https://github.com/jump-dev/HiGHS.jl/blob/1754060e4b7d756749686a9b05583136e418efdd/src/gen/libhighs.jl#L208-L221

image

It'd be nicer if `the HiGHS compilation date` was a properly punctuated sentence :smile:

**Comment by jajhall** [comment]
OK I see that you want a proper sentence following `@returns`, rather than the completion of a sentence beginning with "Returns" :-)

**Comment by odow** [comment]
I've split the `void` issue: https://github.com/ERGO-Code/HiGHS/pull/1201


The other reason to improve these is that we could add the docstrings into the documentation. Here's something I mocked up locally:

image

(With the old docstrings available in HiGHS.jl)

**Comment by jajhall** [comment]
Have you finished?

**Comment by odow** [comment]
Yeah finished for now. Although once https://github.com/ERGO-Code/HiGHS/pull/1203 is merged and we can more easily review the formatting I might have another go.

So either merge, or wait for #1203, up to you.
